toolchain/build-xenon-toolchain

```diff
@@ -27,446 +35,461 @@ BUILD_GCC_SECOND=true
-          $EXTR $NEWLIB_FILE | tar xfz - >> $LOGFILE 2>&1 || fail_with_info
-          cat $NEWLIB.diff | patch -p0 >> $LOGFILE 2>&1 || fail_with_info
-          cd build
-          echo -e "Configuring newlib..."
-          ../$NEWLIB/configure --target=$TARGET --prefix=$PREFIX  --enable-multilib --disable-nls \
-            CFLAGS="-DHAVE_BLKSIZE -O2"\
-            --enable-newlib-io-long-long --enable-newlib-io-long-double >> $LOGFILE 2>&1 || fail_with_info
-          echo -e "Building newlib, this could take a while..."
-          make $PARALLEL 2>&1 >> $LOGFILE || fail_with_info
-          make install 2>&1 >> $LOGFILE || fail_with_info
-          cd ..
-          #rm -rf build/*
-          echo -e "Done"
+                echo -e "Applying vmx128 support..."
+                cat $BINUTILS-vmx128.diff | patch -p0 >> $LOGFILE 2>&1 || fail_with_info
+            fi
         fi
+        cd build
+        echo -e "Configuring $BINUTILS..."
+        ../$BINUTILS/configure --target=$TARGET --prefix="$PREFIX"  --enable-multilib --disable-nls --disable-werror >> $LOGFILE 2>&1 || fail_with_info
+        echo -e "Building $BINUTILS, this could take a while..."
+        make $PARALLEL 2>&1 >> $LOGFILE || fail_with_info
+        echo -e "Installing $BINUTILS..."
+        make install 2>&1 >> $LOGFILE || fail_with_info
+        cd ..
+        rm -rf build/*
+        echo -e "Done"
+    fi
 
-        if $BUILD_GCC_SECOND; then
-          # Yes, you need to build gcc again!
-          cd build
-          echo -e "Configuring gcc - 2nd pass..."
-          unset LIBRARY_PATH
-          ../$GCC/configure --target=$TARGET --prefix=$PREFIX --with-libiconv-prefix=/opt/local --with-cpu=cell \
-                  --with-gmp=/opt/local --with-mpfr=/opt/local --disable-decimal-float --disable-libquadmath \
-                  --enable-languages=c,c++ --disable-libssp --disable-libsanitizer --with-newlib --enable-cxx-flags="-G0" \
-                  --disable-libmudflap --disable-nls --disable-shared --disable-linux-futex --enable-altivec \
-                  --disable-libatomic --disable-threads --disable-libgomp --disable-libitm -v >> $LOGFILE 2>&1 || fail_with_info
-          echo -e "Building gcc - 2nd pass, this could take a while..."
-          make $PARALLEL 2>&1 >> $LOGFILE || fail_with_info
-          make install 2>&1 >> $LOGFILE || fail_with_info
-          cd ..
-          rm -rf build/*
-          echo -e "Done"
+    if $BUILD_GCC; then
+        if [ ! -d $GCC ]; then
+        mkdir $GCC
+        echo -e "Extracting $GCC..."
+        $EXTR $GCC.tar.gz | tar xfz - >> $LOGFILE 2>&1 || fail_with_info
+        cat $GCC.diff | patch -p0 >> $LOGFILE 2>&1 || fail_with_info
         fi
+        cd build
+        echo -e "Configuring $GCC..."
+        ../$GCC/configure --target=$TARGET --prefix="$PREFIX" --with-libiconv-prefix=/opt/local -enable-interwork \
+        --enable-languages="c" --without-headers --disable-shared \
```

Comment:
Could you leave this indented 1 level as they were? it makes it easier to separate what belongs together and not at first glance, same thing with the other similar ones where there is multiple lines with commands